tests/components/nest/test_events.py

```diff
@@ -43,16 +45,15 @@ async def test_event_update_event(hass):
         },
     )
 
-    assert len(hass.states.async_all()) == 1
-
     count = 0
 
     def handle_event(event):
```

Comment:
This test runs a little faster if you annotate this method with `@callback`, as it won't run this method in a thread. We also have a test helper: `events = async_capture_events(hass, "nest_event")` that you can import from `tests.common`. `events` will be a list with all events fired. You would then do `assert len(events) == 0` etc.